src/main/scala/nl.knaw.dans.easy.umd/UpdateMetadataDataset.scala

```diff
@@ -61,24 +61,33 @@ object UpdateMetadataDataset extends DebugEnhancedLogging {
       transformer = Transformer(record.streamID, record.xmlTag, record.oldValue, record.newValue)
       newXML = transformer.transform(oldXML)
       _ <- reportChanges(record, oldXML, newXML)
-      _ <- fedora.updateDatastream(record.fedoraID, record.streamID, newXML.toString())
+      _ <- fedora.updateDatastream(record.fedoraID, record.streamID, stringify(newXML))
     } yield ()
   }.recoverWith {
-    case e => Failure(new Exception(s"failed to process: $record, reason: ${e.getMessage}", e))
+    case e => Failure(new Exception(s"failed to process: $record, reason: ${ e.getMessage }", e))
   }
 
-  private def reportChanges(record: InputRecord, oldXML: Elem, newXML: Seq[Node]): Try[Unit] = {
-    if (oldXML == newXML)
-      Failure(new Exception(s"could not find ${record.streamID} <${record.xmlTag}>${record.oldValue}</${record.xmlTag}>"))
-    else Try {
-      val oldLines = new PrettyPrinter(160, 2).format(oldXML).linesIterator.toList
-      val newLines = new PrettyPrinter(160, 2).format(newXML.head).linesIterator.toList
-
-      logger.info(s"old ${record.streamID}: ${compare(oldLines, newLines)}")
-      logger.info(s"new ${record.streamID}: ${compare(newLines, oldLines)}")
+  private def reportChanges(record: InputRecord, oldXML: Elem, newXML: Seq[Node]): Try[Unit] = Try {
+    (oldXML, newXML) match {
+      case (o, n) if o == n && record.oldValue == "EMPTY" => logger.error(s"This element was already present in the xml, no changes made")
+      case (o, n) if o == n => logger.error(s"could not find ${ record.streamID } <${ record.xmlTag }>${ record.oldValue }</${ record.xmlTag }>")
+      case _ =>
+        val oldLines = new PrettyPrinter(160, 2).format(oldXML).linesIterator.toList
+        val newLines = new PrettyPrinter(160, 2).format(newXML.head).linesIterator.toList
+        logger.info(s"old ${ record.streamID }: ${ compare(oldLines, newLines) }")
+        logger.info(s"new ${ record.streamID }: ${ compare(newLines, oldLines) }")
     }
   }
 
+  private def stringify(newXML: Seq[Node]): String = {
+    val xml = newXML.toString()
+    // In some cases Transformer.transform gives the new xml in a List (e.g. when a new node is ADDED and the parent element is the ROOT element, e.g. DC)
```

Comment:
I'm not sure if this is something that should be corrected for here, or whether this is something that should be fixed in the transformer you wrote in another file...

Reply:
I have looked for a long time in the Transformer, but I couldn't find any way to detect when a List would be returned and when not.